spring-core/src/main/java/org/springframework/core/io/buffer/DefaultDataBuffer.java

```diff
@@ -317,6 +323,43 @@ private void write(ByteBuffer source) {
 		this.writePosition += length;
 	}
 
+	@Override
+	public DataBuffer write(CharSequence charSequence, Charset charset) {
+		Assert.notNull(charSequence, "'charSequence' must not be null");
+		Assert.notNull(charset, "'charset' must not be null");
+		CharsetEncoder charsetEncoder = charset.newEncoder()
```

Comment:
Calling `charset.encode` allocates a `ByteBuffer` (inefficient?) but uses `ThreadLocalCoders.encoderFor` to get a `CharsetEncoder` from a local cache. Because `ThreadLocalCoders` is an internal API, we can't use it here.